src/argilla/client/feedback.py

```diff
@@ -672,6 +680,209 @@ def from_argilla(
+                        f" following types are supported: {list(FIELD_TYPE_TO_PYTHON_TYPE.keys())}"
+                    )
+                features[field.name] = Value(dtype="string", id="field")
+                if field.name not in dataset:
+                    dataset[field.name] = []
+            for question in self.questions:
+                # TODO(alvarobartt): if we constraint ranges from 0 to N, then we can use `ClassLabel` for ratings
+                features[question.name] = Sequence(
+                    {
+                        "user_id": Value(dtype="string"),
+                        "value": Value(dtype="string" if question.settings["type"] == "text" else "int32"),
+                    },
+                    id="question",
+                )
+                if question.name not in dataset:
+                    dataset[question.name] = []
+            features["external_id"] = Value(dtype="string", id="external_id")
+            dataset["external_id"] = []
+
+            for record in self.records:
+                for field in self.fields:
+                    dataset[field.name].append(record["fields"][field.name])
+                for question in self.questions:
+                    dataset[question.name].append(
+                        [
+                            {"user_id": r["user_id"], "value": r["values"][question.name]["value"]}
+                            for r in record["responses"]
+                        ]
+                        or None
+                    )
+                dataset["external_id"].append(record["external_id"] or None)
+
+            return Dataset.from_dict(
+                dataset,
+                features=Features(features),
+            )
+        raise ValueError(f"Unsupported format '{format}'.")
+
+    def push_to_huggingface(self, repo_id: str, generate_card: Optional[bool] = True, *args, **kwargs) -> None:
+        """Pushes the `FeedbackDataset` to the HuggingFace Hub. If the dataset has been previously pushed to the
+        HuggingFace Hub, it will be updated instead. Note that some params as `private` have no effect at all
+        when a dataset is previously uploaded to the HuggingFace Hub.
+
+        Args:
+            repo_id: the ID of the HuggingFace Hub repo to push the `FeedbackDataset` to.
+            generate_card: whether to generate a dataset card for the `FeedbackDataset` in the HuggingFace Hub. Defaults
+                to `True`.
+            *args: the args to pass to `datasets.Dataset.push_to_hub`.
+            **kwargs: the kwargs to pass to `datasets.Dataset.push_to_hub`.
+        """
+        from huggingface_hub import DatasetCard, HfApi
+
+        if len(self) < 1:
+            raise ValueError(
+                "Cannot push an empty `rg.FeedbackDataset` to the HuggingFace Hub, please make sure to add at least one record, via the method `add_records`."
+            )
+
+        hfds = self.format_as("datasets")
```

Comment:
If we move the previous referred code block into an private method, we can use it directly here
```suggestion
        hfds = self._format_as_datasets()
```

Reply:
So I left it like `format_as` to basically allow users to e.g. pull from Argilla, format it as they want (i.e. `datasets`), and then use it, without having to push anything, just to format the current dataset records in the format they want. The idea is to extend the `format_as` method to support `pandas`, `numpy`, etc.